### ISSUE TITLE
typing partial mapping in insert mode

### ISSUE THREAD

**Issue by ratijas** [vimperator/vimperator-labs#543]
##### Issue type:

- Bug report

##### Version:

```
3.13.1 home-compiled
```

##### Description:

typing partial mapping while in `INSERT` mode differs from that of `vim`.  `vim` actually inserts printed character into buffer (or makes it look like inserted) waiting for the next; then if mapping success it clears chars from buffer and executes mapping, otherwise it will timeout soon leaving chars in a buffer.  on the other hand, vimperator won't ever timeout waiting until either mapping is complete or irrelevant character typed instead — latter will normally send all chars into buffer at once.

problem: can not type only beginning of the mapping as normal input, e.g. can not finish text with sequence that is part of map.

##### Steps to reproduce:

```
imap jk <Esc>
imap kj <Esc>
```
type `k` or `j` in any input field

**Comment by timss** [comment]
Able to reproduce.

As a possible workaround, I suggest using mappings with modifiers in insert mode. For example `<C-j>` (note: this is a default firefox keybind for going to `about:home`.

**Comment by ratijas** [comment]
yeah… but the point was that `jk` / `kj` is amazingly easy to hit in one keystroke without any modifiers.

**Comment by ding3my** [comment]
@ratijas We are the same.
http://stackoverflow.com/questions/41234376/how-to-input-a-character-j-after-mapping-jj-to-esc-in-vimperator
